sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -0,0 +1,731 @@
+        """
+        self.map_['source']     = 'srl caltech'
+        self.map_['instrument'] = 'stereo/sit'
+        self.map_['phyobs']     = 'atomic intensities'
+        self.map_['provider']   = 'solar terrestrial relations observatory '
+
+    @classmethod
+    def _can_handle_query(cls, *query, **kwargs):
+        """
+        Answers whether client can service the query.
+        Parameters
+        ----------
+        query : list of query objects
+        Returns
+        -------
+        boolean
+        answer as to whether client can service the query
+        
+        """
+        chkattr =  ['Time', 'Instrument']
+        chklist =  [x.__class__.__name__ in chkattr for x in query]
+        for x in query:
+            if x.__class__.__name__ == 'Instrument' and x.value == 'stereo/sit':
+                return all(chklist)
+        return False
+
+
+
+class PLASTICClient(GenericClient):
+
+    def _get_url_for_timerange(cls,timerange, stereo_spacecraft = 'ahead', duration_of_average = 10 * u.min):
+        """
+        Returns list of URLS to STEREO PLASTIC data files corresponding to value of input timerange.
+        URL Source : http://stereo-ssc.nascom.nasa.gov/data/ins_data/plastic/level2/Protons/ASCII/
+
+        The earliest data available is from 2007.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+            Example value -  TimeRange('2007-02-14','2014-12-17')   
+
+        stereo_spacecraft: string   
+            Default value - ahead
+            Possible values - ahead, behind    # corresponding to spacecraft location
+
+
+        duration_of_average: string
+        Default value - 10 * u.min
+            Possible values - 1 * u.min, 10 * u.min, 1 * u.h      
+            #corresponding to duration over which data is averaged
+
+        Returns
+        -------
+        urls : list
+            list of URLs corresponding to the requested time range
+
+        Examples
+        --------
+        >>> from sunpy.time.timerange import TimeRange
+        >>> from sunpy.net.vso.attrs import Time, Instrument
+        >>> import sunpy.net.dataretriever.sources.stereo as stereo
+        >>> LCClient = stereo.PLASTICClient()
+
+        >>> qr1 = LCClient.query(Time(TimeRange('2012/10/27', '2012/11/27')), Instrument('stereo/plastic'), stereo_spacecraft = 'ahead',
+                                            duration_of_average = 10 * u.min)
+        >>> res = LCClient.get(qr1)
+        >>> download_list = res.wait()
+
+        References
+        ----------
+        | http://stereo-ssc.nascom.nasa.gov/data/ins_data/plastic/level2/Protons/ASCII/
+
+        """
+
+        possible_spacecraft = ['ahead', 'behind']
+        possible_duration   = [1 * u.min, 10 * u.min, 1 * u.h ]
+
+        dict_spacecraft = { 'ahead': 'A', 'behind': 'B' }
+        dict_duration   = { 1 * u.min: '1min', 10 * u.min: '10min', 1 * u.h: '1hr'}
```

Comment:
I think there's a bit of redundancies between the list and dictionaries here (also in other files).  You can check for the keys as if they were lists.